Applications/Spire/Include/Spire/Ui/TableModel.hpp

```diff
@@ -113,12 +114,33 @@ namespace Details {
       virtual boost::signals2::connection connect_operation_signal(
         const typename OperationSignal::slot_type& slot) const = 0;
 
+      /**
+       * Applies a callable to the underlying operation.
+       * @param f The callable to apply.
+       */
+      template<typename F>
+      void visit(const Operation& operation, F&& f) const;
```

Comment:
Prefer free functions over member functions:

https://github.com/isocpp/CppCoreGuidelines/blob/master/CppCoreGuidelines.md#c4-make-a-function-a-member-only-if-it-needs-direct-access-to-the-representation-of-a-class

Reply:
If we will define several visit functions for the different models in the future, will there be a risk of naming confict?

Reply:
No, C++ uses ADL to resolve naming conflicts.

Reply:
Oh, I got it. When the visit function template is instantiated, the function will has different parameters for the different models.